src/program/settings/models.py

```
@@ -14,6 +15,19 @@
 )
 
 
+def validate_empty_or_url(v: Any) -> str:
+    if isinstance(v, str):
+        if v == "":
+            return v
+        if not v.startswith(("http://", "https://", "socks5://", "socks5h://")):
+            raise ValueError("Must be a valid URL or empty string")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)